addons/website/views/website_templates.xml

```diff
@@ -259,1105 +259,1190 @@
+        <attribute name="t-valuef" add="text-center" separator=" "/>
     </xpath>
 </template>
 
-<template id="template_header_default_align_right" inherit_id="website.template_header_default" active="False">
-    <xpath expr="//t[@t-set='_nav_class']" position="replace">
-        <t t-set="_nav_class" t-valuef="ms-auto"/>
+<template id="template_header_mobile_align_right" inherit_id="website.template_header_mobile" active="False">
+    <xpath expr="//t[@t-call='website.navbar_nav']" position="inside">
+        <t t-set="_nav_class" t-valuef="text-end"/>
+    </xpath>
+    <xpath expr="//t[@t-call='website.placeholder_header_text_element']" position="inside">
+        <t t-set="_div_class" t-valuef="mt-2 pt-2 border-top text-end"/>
+    </xpath>
+    <xpath expr="//t[@t-call='website.header_social_links_no_color']" position="inside">
+        <t t-set="_div_class" t-valuef="mt-2 pt-2 border-top text-end"/>
+    </xpath>
+    <xpath expr="//t[@t-set='dropdown_link_class']" position="attributes">
+        <attribute name="t-valuef"/>
+    </xpath>
+    <xpath expr="//t[@t-call='website.submenu']//t[@t-set='dropdown_class']" position="attributes">
+        <attribute name="t-valuef" add="text-end" separator=" "/>
     </xpath>
 </template>
 
-<!-- "Hamburger" template -->
-<template id="template_header_hamburger" inherit_id="website.layout" name="Template Header Hamburger" active="False">
+<!-- "default" template -->
+<template id="template_header_default" inherit_id="website.layout" name="Template Header Default" active="True">
     <xpath expr="//header//nav" position="replace">
         <t t-call="website.navbar">
-            <t t-set="_navbar_expand_class" t-value="''"/>
-            <div id="top_menu_container" class="container">
+            <t t-set="_navbar_classes" t-valuef="d-none d-lg-block py-3 shadow-sm"/>
+
+            <div id="o_main_nav" class="container">
                 <!-- Brand -->
                 <t t-call="website.placeholder_header_brand">
-                    <t t-set="_link_class" t-valuef="me-lg-auto me-0"/>
+                    <t t-set="_link_class" t-valuef="me-4"/>
+                </t>
+                <!-- Navbar -->
+                <t t-call="website.navbar_nav">
+                    <t t-set="_nav_class" t-valuef="me-auto"/>
+
+                    <!-- Menu -->
+                    <t t-foreach="website.menu_id.child_id" t-as="submenu">
+                        <t t-call="website.submenu">
+                            <t t-set="item_class" t-valuef="nav-item"/>
+                            <t t-set="link_class" t-valuef="nav-link"/>
+                        </t>
+                    </t>
                 </t>
-                <ul class="nav navbar-nav navbar-expand d-none d-sm-block">
+                <!-- Extra elements -->
+                <ul class="navbar-nav align-items-center gap-2 flex-wrap justify-content-end ps-3">
```

Comment:
I fixed this using `flex-shrink-0` but it creates an overflow when the viewport is close to the "medium" breakpoint.
I can't fix this with `flex-xx-wrap` otherwise the main navigation will expand and take up half the header.

This behavior could indicate to the user that there are too many elements in the navigation. But I'm not sure this is the right solution.

What do you think?
![Capture d’écran 2023-06-19 à 15 53 24](https://github.com/odoo/odoo/assets/80679690/3e35453f-241b-4844-a9d7-a03a23c4fc68)